goal-planning/goal-planning-frontend
# Feat / Snack Bar Message Notification

## Changes
1. Install Ant Design via `npm`.
2. Configure antd `message` component utilization onto Goal Planning's top level architecture for ease of use and efficiency.
3. Create a message slice for the handling the state reducer logic on the antd `message` component.

## Purpose
Allow better user experience by providing indicator notifications.

## Approach
Install `antd` via `npm install antd` and configure it to work on a top level feature accessible to all child components of the application. In this case, applying antd snack bar's `contextHolder` to the `App.js` file. Then create a redux message slice to handle the snack bar behavior by listening to the message slice reducers that are monitoring specified states in the applications (in our case here, the auth states which are triggered by login, register, and logout async thunk functions). Then finally, apply a `useEffect` code that will show the snack bar message every time we get a reducer action resolved from the message slice.

<img width="703" alt="Screenshot 2023-06-10 at 2 21 25 PM" src="https://github.com/goal-planning/goal-planning-frontend/assets/31714174/d3daec97-946a-4646-84c4-71a5868449d7">

## Testing Steps
1. Pull down code
2. Run the application
3. Test out the registration form. Verify we get a snack bar message for our submissions.
4. Test out the login form. Verify we get a snack bar message for our submissions.
5. Test the logout link. Verify we get a snack bar message when logging out.

## Learning
More understanding of utilizing redux-toolkit. Docs: https://redux-toolkit.js.org/api/createAsyncThunk
Integrating antd `message` into top level component. Docs: https://ant.design/components/message

Closes #61
